app/controllers/hostgroups_controller.rb

```diff
@@ -85,11 +85,11 @@ def puppetclass_parameters
   def environment_selected
     return not_found unless (@environment = Environment.find(params[:environment_id])) if params[:environment_id].to_i > 0
 
-    @hostgroup ||= Hostgroup.new
+    @hostgroup = Hostgroup.find_by_id(params[:hostgroup_id]) || Hostgroup.new
```

Comment:
The bug I am seeing is that when I select an environment on the hostgroups page, it's showing all puppet classes for every puppet environment:

https://youtu.be/jxkQ4ALC9MQ

It looks like the UI is sending but hostgroup_id Hostgroups#environment_selected is not doing anything with it. 

I looked at adding `before_action :find_resource` but it looks like it calls `find` and I am not 100% sure that hostgroup_id is always being sent as environment_selected also accepts environment_id as well.